components/binding/internal/webhook/pod/mutate/mutate.go

```diff
@@ -55,7 +52,7 @@ func (h *MutationHandler) Handle(ctx context.Context, req admission.Request) adm
 		return admission.Allowed("pod has no any assigned bindings. action not taken.")
 	}
 
-	bindings, err := h.findBindings(ctx, bindingsName, req.Namespace)
+	bindings, err := webhook.FindBindings(ctx, h.client, bindingsName, req.Namespace)
```

Comment:
`h.client` is used only to pass to "webhook finds" methods, so maybe all "finds" method should be part of webhook.client 
```go
package webhook

type Client struct {
  client  client.Client
}

func (c *client) FindSecret()
func (c *client) FindConfigMap()
func (c *client) FindBindings()
```
and then you can pass webhook.client to the `MutationHandler` and use:
```go
bindings, err := h.client.FindBindings(ctx, bindingsName, req.Namespace)
```